packages/layout/src/graph-iface.ts

```diff
@@ -1084,10 +1084,10 @@ function addEdge(graph: IGraphClassic,
 
     let visualEdge: VisualRelationship | VisualProfileRelationship | null = null;
     if(visualModel !== null) {
-        if(edgeToAddKey === "outgoingEdgeProfileEdges") {
+        if(edgeToAddKey === "outgoingProfileEdges") {
             // TODO: Again the ID of semantic model instead of the visual one
             const visualEntityForEdge = visualModel.getVisualEntityForRepresented(id);
-            if(isVisualProfileRelationship(visualEntityForEdge)) {
+            if(isVisualRelationship(visualEntityForEdge)) {
```

Comment:
This was the main issue why the layouting of profile edges didn't work. The edge has type "visual-relationship" - Not sure if it is wrong or I just misunderstood it @skodapetr

Reply:
The naming should be improved the isVisualRelationship represents a visual representation of a relationship that exists as an entity. The VisualProfileRelationship is specialization of VisualNodeRelationship. VisualNodeRelationship is build to represent a relationship defined by an entity. As such it is absent of a semantic entity representing a relationship. As a result there should be `or` for both of them.